### PR TITLE
Created a URL shortener service

### PR DESCRIPTION
**Pull request by ulitol97**
This PR targets #117.

I've implemented a new endpoint in the server in `/api/permalink/generate`. It connects to [bit.ly](https://bitly.com/) to generate shortened URLs.

Clients may query this endpoint sending a `GET` request containing a `url` parameter that contains the link to be shortened.

For example, a call to `/api/permalink/generate?url=https://github.com` would return [https://bit.ly/3dnDVGh](https://bit.ly/3dnDVGh)

### CODE REVIEW

**Comment by mistermboy** [comment]
@ulitol97  Is [bit.ly](https://bitly.com/) a free service?

File: modules/server/src/main/scala/es/weso/server/PermalinkService.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    // Query URL shortener API
+    case GET -> Root / `api` / "permalink" / "generate" :?
+      UrlParam(url) =>
+      // Request the shortened URL
+      try {
+        val res = Http(urlShortenerEndpoint).postData(s"""{ "long_url": "$url" }""")
+          .header("Content-Type", "application/json")
+          .header("Accept", "application/json")
+          .header("Authorization", "8e4c5fc54b280e5bc969fc0ef7a0f759317b7c64")
```
Comment (thewilly) [inline modules/server/src/main/scala/es/weso/server/PermalinkService.scala:29]: @ulitol97 is this the bit.ly authorization key?

**Comment by ulitol97** [comment]
> @ulitol97 Is [bit.ly](https://bitly.com/) a free service?

Yes. It seemed to be the best option out of all the others. It only needs to be linked to an account. I'm still thinking about using [TinyUrl](https://gist.github.com/MikeRogers0/2907534), but that looks more like a hack that may stop working any moment.

> @ulitol97 is this the bit.ly authorization key?

Yes, that is how it is linked to an account and why this functionality is not in the clients. I'm planning on hiding it more.

**Comment by ulitol97** [comment]
I have eventually changed the shortener service to https://cutt.ly/, which had an unlimited free plan: #119.

**Pull request merged**